app/models/validations/tenancy_validations.rb

```diff
@@ -41,4 +47,8 @@ def validate_joint_tenancy(record)
       record.errors.add :hhmemb, I18n.t("validations.tenancy.joint_more_than_one_member")
     end
   end
+
+  def min_tenancy_length(record)
+    record.needstype == 2 || record.renttype == 3 ? 1 : 2
```

Comment:
```suggestion
    record.is_supported_housing? == 2 || record.renttype == 3 ? 1 : 2
```

I try to use our methods if we can. Apart from being easier to read, it helps us if we ever migrate values across forms.